src/webgpu/api/operation/render_pass/resolve.spec.ts

```diff
@@ -0,0 +1,204 @@
+        resolveTargets.push(
+          t.device.createTexture({
+            format: kFormat,
+            size: {
+              width: kResolveTargetSize,
+              height: kResolveTargetSize,
+              depth: t.params.resolveTargetBaseArrayLayer + 1,
+            },
+            sampleCount: 1,
+            mipLevelCount: t.params.resolveTargetBaseMipLevel + 1,
+            usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+          })
+        );
+
+        // Clear to black for the load operation. After the draw, the top left half of the attachment
+        // will be white and the bottom right half will be black.
+        renderPassColorAttachmentDescriptors.push({
+          attachment: colorAttachments[i].createView(),
+          loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 0.0 },
+          storeOp: t.params.storeOperation,
+          resolveTarget: resolveTargets[numResolveTargets].createView({
+            baseMipLevel: t.params.resolveTargetBaseMipLevel,
+            baseArrayLayer: t.params.resolveTargetBaseArrayLayer,
+          }),
+        });
+        numResolveTargets++;
+      } else {
+        renderPassColorAttachmentDescriptors.push({
+          attachment: colorAttachments[i].createView(),
+          loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 0.0 },
+          storeOp: t.params.storeOperation,
+        });
+      }
+    }
+
+    const encoder = t.device.createCommandEncoder();
+
+    const pass = encoder.beginRenderPass({
+      colorAttachments: renderPassColorAttachmentDescriptors,
+    });
+    pass.setPipeline(pipeline);
+    pass.draw(3, 1, 0, 0);
+    pass.endPass();
+    t.device.defaultQueue.submit([encoder.finish()]);
+
+    for (let i = 0; i < numResolveTargets; i++) {
+      // Test top left pixel, which should be {255, 255, 255, 255}.
+      t.expectSinglePixelIn2DTexture(
+        resolveTargets[i],
+        kFormat,
+        { x: 0, y: 0 },
+        {
+          exp: new Uint8Array([0xff, 0xff, 0xff, 0xff]),
+          slice: t.params.resolveTargetBaseArrayLayer,
+          layout: { mipLevel: t.params.resolveTargetBaseMipLevel },
+        }
+      );
+
+      // Test bottom right pixel, which should be {0, 0, 0, 0}.
+      t.expectSinglePixelIn2DTexture(
+        resolveTargets[i],
+        kFormat,
+        { x: kSize - 1, y: kSize - 1 },
+        {
+          exp: new Uint8Array([0x00, 0x00, 0x00, 0x00]),
+          slice: t.params.resolveTargetBaseArrayLayer,
+          layout: { mipLevel: t.params.resolveTargetBaseMipLevel },
+        }
+      );
+
+      // Test top right pixel, which should be {127, 127, 127, 127} due to the multisampled resolve.
+      t.expectSinglePixelIn2DTexture(
+        resolveTargets[i],
+        kFormat,
+        { x: kSize - 1, y: 0 },
+        {
+          exp: new Uint8Array([0x7f, 0x7f, 0x7f, 0x7f]),
+          slice: t.params.resolveTargetBaseArrayLayer,
+          layout: { mipLevel: t.params.resolveTargetBaseMipLevel },
+        }
+      );
```

Comment:
nit: I know we have `storeOp.spec.ts` but I think it might be valuable here to check a pixel in the non-resolved attachment to make sure that one was correctly written/cleared. Particularly if there was a resolve target and storeOp were `clear`, I could see there being bugs where the implementation wrongly does a store.

Reply:
I agree it would be nice to test - but I'm not seeing a non-complex way to do this.
1. We can't copy from a multisampled texture.
2. We can't render to both multisampled and single-sampled textures in the same pass.

So to check this for correctness, I guess we could render multisampled without resolve, then do a subsequent renderpass that samples from the multisampled texture and renders it into a single sampled texture. WDYT?

Reply:
Another option is to do a texel fetch in a compute shader, which might be more complex than we want for this test. I'd be okay with a TODO for now, and when we implement better helpers to check texture contents we can improve. @kainino0x WDYT?

Reply:
I do think we should test this. Let's leave a TODO.

We're starting to run into more situations where we need to be able to test values that can't be copied. Someone should own an item to implement this as completely as possible (I guess it can be me) before we accumulate too much tech debt.

Reply:
(I'm guessing such a helper would need to be glsl-dependent for now, unfortunately.)

Reply:
Added TODO at the top.